plugins/discovery-gce/src/main/java/org/elasticsearch/cloud/gce/GceComputeServiceImpl.java

```diff
@@ -207,15 +206,13 @@ public Void run() throws IOException {
                 refreshInterval = TimeValue.timeValueSeconds(credential.getExpiresInSeconds() - 1);
             }
 
-            boolean ifRetry = settings.getAsBoolean(Fields.RETRY, true);
-            Compute.Builder builder = new Compute.Builder(getGceHttpTransport(), gceJsonFactory, null)
-                    .setApplicationName(Fields.VERSION);
+            Compute.Builder builder = new Compute.Builder(getGceHttpTransport(), gceJsonFactory, null).setApplicationName(VERSION);
 
-            if (ifRetry) {
-                int maxWait = settings.getAsInt(Fields.MAXWAIT, -1);
+            if (RETRY_SETTING.get(settings)) {
+                TimeValue maxWait = MAXWAIT_SETTING.get(settings);
                 RetryHttpInitializerWrapper retryHttpInitializerWrapper;
 
-                if (maxWait > 0) {
+                if (maxWait.getMillis() > 0) {
                     retryHttpInitializerWrapper = new RetryHttpInitializerWrapper(credential, maxWait);
                 } else {
                     retryHttpInitializerWrapper = new RetryHttpInitializerWrapper(credential);
```

Comment:
This seems to conflict with the documentation which says that "A negative value means that it will wait indefinitely.". But this waits just for 15 minutes. And it also does the same for 0s.